Sources/Brave/Frontend/Browser/PageData.swift

```diff
@@ -98,18 +98,26 @@ struct PageData {
       }
     }
     
+    let allEngineScriptTypes = await makeAllEngineScripts(for: domain)
+    return userScriptTypes.union(allEngineScriptTypes)
+  }
+  
+  func makeMainFrameEngineScriptTypes(domain: Domain) async -> Set<UserScriptType> {
+    return await adBlockStats.makeEngineScriptTypes(frameURL: mainFrameURL, isMainFrame: true, domain: domain)
+  }
+  
+  func makeAllEngineScripts(for domain: Domain) async -> Set<UserScriptType> {
     // Add engine scripts for the main frame
-    userScriptTypes = userScriptTypes.union(
-      adBlockStats.makeEngineScriptTypes(frameURL: mainFrameURL, isMainFrame: true, domain: domain)
-    )
+    async let engineScripts = adBlockStats.makeEngineScriptTypes(frameURL: mainFrameURL, isMainFrame: true, domain: domain)
```

Comment:
This async let is also not very useful for the same reason `makeEngineScriptTypes` will block main actor anyways and thus the calls below that also make calls to `makeEngineScriptTypes` will not execute until after this is done

Reply:
So as I said above I made the method non `@MainActor` so this should also allow this to be in parallel